app/models/user.rb

```diff
@@ -2,6 +2,21 @@ class User < ApplicationRecord
   has_secure_password
 
   has_many :articles, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  # Will return an array of follows for the given user instance
+  has_many :received_follows, foreign_key: :followed_user_id, class_name: "Follow"
+
+  # Will return an array of users who follow the user instance
+  has_many :followers, through: :received_follows, source: :follower
```

Comment:
what does `through`?

Reply:
What it does is call the received_followers method that was defined previously and it obtains a temporary array by calling it which contains the union of the follwer_user_id and the user_id and thus passing it with source to follower thus obtaining a new array of instances of the user's followers in question